src/librustdoc/html/render/print_item.rs

```diff
@@ -278,11 +278,10 @@ pub(super) fn print_item(cx: &Context<'_>, item: &clean::Item, buf: &mut Buffer)
     // Render notable-traits.js used for all methods in this module.
     let mut types_with_notable_traits = cx.types_with_notable_traits.borrow_mut();
     if !types_with_notable_traits.is_empty() {
-        write!(
-            buf,
-            r#"<script type="text/json" id="notable-traits-data">{}</script>"#,
+        write!(buf, r#"<script type="text/json" id="notable-traits-data">{}</script>"#, {
+            #[allow(rustc::potential_query_instability)]
```

Comment:
This is technically ok, as `notable_traits_json` sorts the items, but just looking at this code, that's not clear why.

Given that this is the [only callsite](https://github.com/search?q=repo%3Arust-lang%2Frust%20notable_traits_json&type=code), it'd be better to pass a `FxHashMap` (instead of an opaque `impl Iterator`, so the allow can be next to a comment explaining why it's ok. That way all that reasoning about why this is correct stays local.

Reply:
I agree, fixed.
(I used the `#[expect]` attribute because it's cool and I hadn't had a chance to use it yet, not sure if it should just be `#[allow]`)